zebrad/src/commands/start/sync.rs

```diff
@@ -219,27 +212,24 @@ where
                         }
                     }
 
-                    // Hashes will be empty if we know about all the blocks in the response.
-                    if first_unknown.is_none() {
-                        tracing::debug!("ObtainTips: all hashes are known");
-                        continue;
-                    }
-                    let first_unknown = first_unknown.expect("already checked for None");
-                    tracing::debug!(
-                        first_unknown,
-                        "found index of first unknown hash in response"
-                    );
+                    tracing::debug!(hashes.len = ?hashes.len(), ?first_unknown);
 
-                    let unknown_hashes = &hashes[first_unknown..];
+                    let unknown_hashes = if let Some(index) = first_unknown {
+                        &hashes[index..]
+                    } else {
+                        continue;
+                    };
                     let new_tip = *unknown_hashes
                         .last()
-                        .expect("already checked that unknown hashes isn't empty");
+                        .expect("first unknown index < hashes.len() so slice is nonempty");
```

Comment:
Issue: enumerate can't return an index that is greater than the length of the iterator.

```suggestion
                        .expect("first unknown exists, so slice is nonempty");
```

Reply:
Right, but that invariant (`index < hashes.len()` is why the slice is nonempty, which is why it's in the `.expect` -- the index existing isn't enough, since it could be `Some(index)` with `index > len` and the slice would be empty.  (But this code is removed when changing to pairs of hashes anyways).